src/order/succ_pred/relation.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2022 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import order.succ_pred.basic
+/-!
+# Relations on types with a `succ_order`
+
+This file contains properties about relations on types with a `succ_order`
+and their closure operations (like the transitive closure).
+-/
+
+open set relation succ_order function
+
+variables {α : Type*} [linear_order α] [succ_order α] [is_succ_archimedean α] [no_max_order α]
```

Comment:
I am pretty sure you can do without `no_max_order`.
```suggestion
variables {α : Type*} [linear_order α] [succ_order α] [is_succ_archimedean α]
```

Reply:
You're right! I have to do a little extra step in the proof, but it works.